### PR TITLE
config: Move cmdline config to domain.cfg

### PR DESCRIPTION
**Pull request by Deedone**
Move cmdline configuration from u-boot to domain.cfg. This will allow for each domain to have its own cmdline configuration.

### CODE REVIEW

**Comment by Deedone** [comment]
https://github.com/xen-troops/u-boot/pull/21

**Comment by xakep-amatop** [comment]
should you add changes for h3ulcb board too?

**Comment by Deedone** [comment]
> should you add changes for h3ulcb board too?

Hi, cmdline values are taken from the u-boot config, and it only has values for Salvator and Spider boards.

**Comment by xakep-amatop** [comment]
> > should you add changes for h3ulcb board too?
> 
> Hi, cmdline values are taken from the u-boot config, and it only has values for Salvator and Spider boards.

According to README here: https://github.com/aoscloud/aos_core_zephyr/pull/137#issuecomment-2006136506
it is possible to build aos for `h3ulcb-4x2g`

is it possible that we use the same DomD u-boot for h3ulcb as for salvator board ?

**Comment by Deedone** [comment]
> is it possible that we use the same DomD u-boot for h3ulcb as for salvator board ?

Seems unlikely because the same config explicitly specifies salvator-x device tree for the u-boot itself.

@al1img Hi, could you please clarify where are we getting domd bootargs on h3ulcb builds?

**Comment by lorc** [comment]
@al1img, I believe, this is the internal AOS affairs now

Just tell me when to merge https://github.com/xen-troops/u-boot/pull/21, so there will be no discrepancy between the U-Boot and  AOS APP configurations.

**Comment by al1img** [comment]
> > is it possible that we use the same DomD u-boot for h3ulcb as for salvator board ?
> 
> Seems unlikely because the same config explicitly specifies salvator-x device tree for the u-boot itself.
> 
> @al1img Hi, could you please clarify where are we getting domd bootargs on h3ulcb builds?

AFAIK, we used the same u-boot for both gen3 boards. So, cmdline options should be added to `domd_cfg_h3ulcb.c` too. Also, could you rebase this PR for develop branch?

**Comment by al1img** [comment]
> @al1img, I believe, this is the internal AOS affairs now
> 
> Just tell me when to merge [xen-troops/u-boot#21](https://github.com/xen-troops/u-boot/pull/21), so there will be no discrepancy between the U-Boot and AOS APP configurations.

You can merge it.

File: src/domains/domd/domd_cfg_spider.c [hunk 1]
```diff
@@ -309,6 +309,7 @@ struct xen_domain_cfg domd_cfg = {
     .load_image_bytes  = load_ipl_image,
     .get_image_size    = get_ipl_image_size,
     .image_info        = NULL,
+    .cmdline = "root=/dev/sda4 ro rootwait console=hvc0 clk_ignore_unused pci=pcie_bus_perf vardir.disk=/dev/sda5 opendisk.target=/dev/sda6 opendisk.pkcs11=optee aosupdate.disk=/dev/aosvg/workdirs aosupdate.path=um/update_rootfs aosupdate.selinux_module=/usr/share/selinux/aos/base.pp",
```
Comment (al1img) [inline src/domains/domd/domd_cfg_spider.c:312]: ditto

**Review by al1img — APPROVED** [review]

**Pull request merged**